### ISSUE TITLE
TechDocs Review and Publishing Process Improvements

### ISSUE THREAD

**Issue by AErmie** [bcgov/Cloud-Pathfinder-Azure#201]
With the release of the initial Azure content, and the growing number of users, we need to establish a streamlined process for updating client-facing documentation, team review, and the publishing process.

## Suggested Workflow

1. Members of the Public Cloud team (either Platform Administrators, or Developers) creates branches off of `dev` for content updates
2. Once updates are written, a Pull Request is created for members of the Public Cloud team to self-review before merging into the `dev` branch
3. Weekly/Bi-weekly a Pull Request is created for `dev` into `main` for Pilar Solares to review. Once review is complete, merge into `main` and publish

**Comment by AErmie** [comment]
Briefly discussed with Pilar Solares. 

## Current workflow

My work depends on my capacity and communication from the team hence why I am reaching out here. Usually I get from the team a message about them working on a particular ticket for content and we determine what is needed and when is needed

## Follow up

TODO: Set up a meeting with Pilar to discuss the current process, and opportunities to streamline.

**Comment by AErmie** [comment]
Met with Pilar to discuss the review process.

She needs sufficient lead-time, due to other priorities with other teams as well. Also, if a documentation update is urgent, she would appreciate a heads-up, in order to prioritize it.

It is agreed that if the Public Cloud team self-manages their changes within/to the `dev` branch, then once ready for LIVE, submit a Pull Request (PR) into the `main` branch, and add Pilar for review. This will allow the team to make necessary documentation updates as required, and perform internal content checks, before requesting review from Pilar. It will also ensure there is a single PR for Pilar to review, versus multiple PRs with many little changes.

## Pull Request Checks
A new GitHub Workflow has been created for Pull Requests. This workflow will build the TechDocs container, and also execute the [htmltest](https://github.com/bcgov/devhub-techdocs-publish/blob/main/docs/index.md#link-validation-using-htmltest) to validate the links within the pages.

We've included the `strict_validation: 'true'` parameter, to cause the PR Check to fail if any broken links are discovered.

### Link Exceptions

Due to some links requiring authentication, these will appear as failures in the `htmltest` validation. Therefore we've also included a `.htmltest.yml` file, with the following additional configurations:

```yaml
IgnoreURLs:
    - "https://github.com/bcgov-c"
    - "https://login.nimbus.cloud.gov.bc.ca/"
```

Further, for various external links (ie. Microsoft), that throw errors (though the links themselves are accurate), we've added the following inline ignore code: `{:data-proofer-ignore}`

For more details about this YAML file, please refer to the following documentation: [Configuring htmltest](https://github.com/bcgov/devhub-techdocs-publish/blob/main/docs/index.md#configuring-htmltest)
